test/FunctionalTests/Microsoft.OData.Edm.Tests/Csdl/CsdlReaderTests.cs

```diff
@@ -711,6 +711,105 @@ private static IEdmModel GetEdmModel(string types = "", string properties = "")
+                 "<Function Name=\"TestFunction\">" +
+                   "<Parameter Name=\"requiredParam\" Type=\"Edm.String\" Nullable=\"false\" />" +
+                   "<Parameter Name=\"optionalParam\" Type=\"Edm.String\" Nullable=\"false\">" +
+                       "<Annotation Term=\"Org.OData.Core.V1.OptionalParameter\" />" +
+                   "</Parameter>" +
+                   "<Parameter Name=\"optionalParamWithDefault\" Type=\"Edm.String\" Nullable=\"false\">" +
+                       "<Annotation Term=\"Org.OData.Core.V1.OptionalParameter\">" +
+                         "<Record>" +
+                           "<PropertyValue Property=\"DefaultValue\" String=\"Smith\" />" +
+                         "</Record>" +
+                       "</Annotation>" +
+                   "</Parameter>" +
+                   "<ReturnType Type=\"Edm.String\" Nullable=\"false\" />" +
+                 "</Function>" +
+               "</Schema>" +
+             "</edmx:DataServices>" +
+           "</edmx:Edmx>";
+
+            IEdmModel model;
+            IEnumerable<EdmError> errors;
+
+            bool result = CsdlReader.TryParse(XElement.Parse(expected).CreateReader(), out model, out errors);
+            Assert.True(result);
+            Assert.NotNull(model);
+
+            var function = model.SchemaElements.OfType<IEdmFunction>().First();
+            VerifyOptionalParameter(function);
+        }
+
+        [Fact]
+        public void ParsingOutOfLineOptionalParameterWorks()
+        {
+            string expected =
+           "<?xml version=\"1.0\" encoding=\"utf-16\"?>" +
+           "<edmx:Edmx Version=\"4.0\" xmlns:edmx=\"http://docs.oasis-open.org/odata/ns/edmx\">" +
+             "<edmx:DataServices>" +
+               "<Schema Namespace=\"NS\" xmlns=\"http://docs.oasis-open.org/odata/ns/edm\">" +
+                 "<Function Name=\"TestFunction\">" +
+                   "<Parameter Name=\"requiredParam\" Type=\"Edm.String\" Nullable=\"false\" />" +
+                   "<Parameter Name=\"optionalParam\" Type=\"Edm.String\" Nullable=\"false\" />" +
+                   "<Parameter Name=\"optionalParamWithDefault\" Type=\"Edm.String\" Nullable=\"false\" />" +
+                   "<ReturnType Type=\"Edm.String\" Nullable=\"false\" />" +
+                 "</Function>" +
+                 "<EntityContainer Name=\"Default\">" +
+                   "<FunctionImport Name=\"TestFunction\" Function=\"test.TestFunction\" />" +
+                 "</EntityContainer>" +
+                 "<Annotations Target=\"NS.TestFunction(Edm.String, Edm.String, Edm.String)/optionalParam\" >" +
+                   "<Annotation Term=\"Org.OData.Core.V1.OptionalParameter\" >" +
+                     "<Record />" +
+                  "</Annotation> " +
+                 "</Annotations>" +
+                 "<Annotations Target=\"NS.TestFunction(Edm.String, Edm.String, Edm.String)/optionalParamWithDefault\" >" +

```

Comment:
>\"NS.TestFunction(Edm.String, Edm.String, Edm.String) [](start = 38, length = 53)

should add test cases for NS.TestFunction/optionalParam and NS.TestFunction/optionalParamWithDefault.

Reply:
Added. Thanks.